lib/std/child_process.zig

```diff
@@ -280,22 +280,38 @@ pub const ChildProcess = struct {
     }
 
     fn cleanupAfterWait(self: *ChildProcess, status: u32) !Term {
-        defer {
-            os.close(self.err_pipe[0]);
-            os.close(self.err_pipe[1]);
-        }
-
-        // Write maxInt(ErrInt) to the write end of the err_pipe. This is after
-        // waitpid, so this write is guaranteed to be after the child
-        // pid potentially wrote an error. This way we can do a blocking
-        // read on the error pipe and either get maxInt(ErrInt) (no error) or
-        // an error code.
-        try writeIntFd(self.err_pipe[1], maxInt(ErrInt));
-        const err_int = try readIntFd(self.err_pipe[0]);
-        // Here we potentially return the fork child's error
-        // from the parent pid.
-        if (err_int != maxInt(ErrInt)) {
-            return @errSetCast(SpawnError, @intToError(err_int));
+        defer destroyPipe(self.err_pipe);
+
+        if (builtin.os == .linux) {
+            var fd = [1]std.os.pollfd{std.os.pollfd{
+                .fd = self.err_pipe[0],
+                .events = std.os.POLLIN,
+                .revents = undefined,
+            }};
+
+            // Check if the eventfd buffer stores a non-zero value by polling
+            // it, that's the error code returned by the child process.
+            _ = std.os.poll(&fd, 0) catch unreachable;
```

Comment:
You can just `read()`.

>  If the eventfd counter is zero at the time of the call to read(2), then the call either blocks until the counter becomes nonzero (at which time, the read(2) proceeds as described above) or fails with the error EAGAIN if the file descriptor has been made nonblocking.

Reply:
Yeah, no. The `poll` call is meant to peek into the eventfd buffer as it may be zero if the child returns no error. Using a non blocking is quite cumbersome here as we don't want to event loop to kick in if EINTR is returned by `read`.